### ISSUE TITLE
Project System: Figure out upgrade

### ISSUE THREAD

**Issue by davkean** [dotnet/project-system#34]
_From @davkean on May 13, 2016 2:10_

While most projects don't need to upgrade from VS 2015 -> VS 15, we still have to handle project upgrade from earlier versions.

We basically need to fill out implementations of IFlavoredProjectCompatibilityProvider.IsProjectNeedBeUpgradedAsync to decide whether to delegate onto csproj.

_Copied from original issue: dotnet/roslyn#11285_

**Comment by panopticoncentral** [comment]
@jjmew @davkean Assuming I'm understanding this issue correctly, I think we should close it. SDK-style projects didn't exist before 2017, obviously. If we did support opening non-SDK style projects, I think it would be reasonable to say that we won't support pre-VS 2017 upgrade directly. You could always use the CSProj project system to updated and then switch over. Let me know if I've got this wrong, or you disagree.

**Comment by davkean** [comment]
Yes we would still need this feature. Let's say tomorrow we say "Xamarin" projects automatically open in CPS, if they have upgrade paths we need to be able to handle that situation.

**Comment by jjmew** [comment]
Reopening based on Dave's comments
